core/chains/evm/txmgr/evm_tx_store.go

```diff
@@ -1292,26 +1292,30 @@ func (o *evmTxStore) SaveInProgressAttempt(ctx context.Context, attempt *TxAttem
 	return nil
 }
 
-func (o *evmTxStore) GetNonFatalTransactions(ctx context.Context, chainID *big.Int) (txes []*Tx, err error) {
+func (o *evmTxStore) GetAbandonedTransactionsByBatch(ctx context.Context, chainID *big.Int, enabledAddrs []common.Address, offset, limit uint) (txes []*Tx, err error) {
 	var cancel context.CancelFunc
 	ctx, cancel = o.mergeContexts(ctx)
 	defer cancel()
+
+	var enabledAddrsBytea [][]byte
+	for _, addr := range enabledAddrs {
+		enabledAddrsBytea = append(enabledAddrsBytea, addr[:])
+	}
+
 	err = o.Transaction(ctx, true, func(orm *evmTxStore) error {
```

Comment:
Do we need a transaction for this? Maybe a simple query will do the work.

Reply:
We don't need need a transaction here, I'll remove it.